### PR TITLE
Dev/ameet/automation testing

### PR DESCRIPTION
**Pull request by ameetj85**
Created automation tests for 3 container samples

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ how-to/*/package-lock.json
 **/public/uploads/applogs/*.zip
 
 ./public*/
+package-lock.old.json
```
Comment (obany) [inline .gitignore:18]: No need for this in .gitignore

File: how-to/automation-testing/create-window-js/automation-hooks.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+exports.setup = async (globalVars) => {
```
Comment (obany) [inline how-to/automation-testing/create-window-js/automation-hooks.js:1]: You can remove this file if you are not using hooks

File: how-to/automation-testing/create-window-js/jasmine.reporter.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/create-window-js/jasmine.reporter.json:1]: You can remove this file if you are not using jasmine test runner

File: how-to/automation-testing/create-window-js/jest.reporter.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/create-window-js/jest.reporter.json:1]: You can remove this file if you are not using jest test runner

File: how-to/automation-testing/create-window-js/offline-storage/offline-versions.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (obany) [inline how-to/automation-testing/create-window-js/offline-storage/offline-versions.json:1]: You can remove this folder if you dont want to run in offline mode

File: how-to/automation-testing/create-window-js/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+	},
+	"author": "ameet.jayawant@openfin.co",
+	"license": "SEE LICENSE IN LICENSE.MD",
+	"devDependencies": {
+		"@openfin/automation-cli": "^1.0.0",
+		"@openfin/automation-helpers": "^1.0.0",
+		"@openfin/automation-native": "^1.0.0",
+		"chai": "4.3.6",
+		"jasmine-pretty-html-reporter": "0.2.5",
```
Comment (obany) [inline how-to/automation-testing/create-window-js/package.json:16]: You can remove jasmine and jest reporters

File: how-to/automation-testing/create-window-js/tests/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable linebreak-style */
```
Comment (obany) [inline how-to/automation-testing/create-window-js/tests/index.spec.js:1]: We should try not to disable lint rules as the main ones are there for consistency

File: how-to/automation-testing/use-launch-external-process-js/jasmine.reporter.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/use-launch-external-process-js/jasmine.reporter.json:1]: You can remove this file if you are not using jasmine test runner

File: how-to/automation-testing/use-launch-external-process-js/jest.reporter.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/use-launch-external-process-js/jest.reporter.json:1]: You can remove this file if you are not using jest test runner

File: how-to/automation-testing/use-launch-external-process-js/package.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+	},
+	"author": "ameet.jayawant@openfin.co",
+	"license": "SEE LICENSE IN LICENSE.MD",
+	"devDependencies": {
+		"@openfin/automation-cli": "^1.0.0",
+		"@openfin/automation-helpers": "^1.0.0",
+		"@openfin/automation-native": "^1.0.0",
+		"chai": "4.3.6",
+		"jasmine-pretty-html-reporter": "0.2.5",
```
Comment (obany) [inline how-to/automation-testing/use-launch-external-process-js/package.json:16]: Remove jasmine and jest

File: how-to/automation-testing/use-launch-external-process-js/tests/index.spec.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
+/* eslint-disable linebreak-style */
```
Comment (obany) [inline how-to/automation-testing/use-launch-external-process-js/tests/index.spec.js:1]: Remove eslint disables, fix the code instead

File: how-to/automation-testing/use-window-options-js/automation-hooks.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable linebreak-style */
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/automation-hooks.js:1]: Not needed

File: how-to/automation-testing/use-window-options-js/jasmine.reporter.json [hunk 13]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/jasmine.reporter.json:1]: You can remove this file if you are not using jasmine test runner

File: how-to/automation-testing/use-window-options-js/jest.reporter.json [hunk 14]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/jest.reporter.json:1]: You can remove this file if you are not using jest test runner

File: how-to/automation-testing/use-window-options-js/offline-storage/offline-versions.json [hunk 15]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/offline-storage/offline-versions.json:1]: Remove offline folder if not running on offline mode

File: how-to/automation-testing/use-window-options-js/package.json [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+	},
+	"author": "ameet.jayawant@openfin.co",
+	"license": "SEE LICENSE IN LICENSE.MD",
+	"devDependencies": {
+		"@openfin/automation-cli": "^1.0.0",
+		"@openfin/automation-helpers": "^1.0.0",
+		"@openfin/automation-native": "^1.0.0",
+		"chai": "4.3.6",
+		"jasmine-pretty-html-reporter": "0.2.5",
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/package.json:16]: Remove jasmine and jest

File: how-to/automation-testing/use-window-options-js/tests/index.spec.js [hunk 17]
```diff
@@ -0,0 +1,154 @@
+/* eslint-disable linebreak-style */
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/tests/index.spec.js:1]: Remove disables and fix code instead

**Review by obany — CHANGES_REQUESTED** [review]
A few comments, mostly around things that can be removed

File: how-to/automation-testing/create-window-js/automation-hooks.js [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+	globalVars.startTime = Date.now();
+};
+
+exports.teardown = async (globalVars) => {
+	console.log(`Global Teardown`);
+	console.log(`Time taken`, Date.now() - globalVars.startTime);
+};
+
+exports.beforeEach = async (globalVars, currentTestName) => {
+	// console.log("Before", currentTestName);
```
Comment (johnman) [inline how-to/automation-testing/create-window-js/automation-hooks.js:12]: Could we remove the commented out code or uncomment it?

File: how-to/automation-testing/create-window-js/tests/index.spec.js [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
+	it('The url should be set', async () => {
+		providerWindowUrl = await WebDriver.getUrl();
+		expect(providerWindowUrl).not.be.undefined;
+	});
+
+	it('The runtime version should be set', async () => {
+		const fin = await OpenFinProxy.fin();
+		const version = await fin.System.getVersion();
+		expect(version).to.equal('26.102.70.16');
```
Comment (johnman) [inline how-to/automation-testing/create-window-js/tests/index.spec.js:42]: Should we pull this from the manifest if we have the fin api? That way the runtime version isn't hardcoded and isn't something we will miss in the future?

File: how-to/automation-testing/use-launch-external-process-js/automation-hooks.js [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+	globalVars.startTime = Date.now();
+};
+
+exports.teardown = async (globalVars) => {
+	console.log(`Global Teardown`);
+	console.log(`Time taken`, Date.now() - globalVars.startTime);
+};
+
+exports.beforeEach = async (globalVars, currentTestName) => {
+	// console.log("Before", currentTestName);
```
Comment (johnman) [inline how-to/automation-testing/use-launch-external-process-js/automation-hooks.js:13]: Same as before. Can we uncomment or remove?

File: how-to/automation-testing/use-launch-external-process-js/tests/index.spec.js [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+
+	it('The url should be set', async () => {
+		providerWindowUrl = await WebDriver.getUrl();
+		expect(providerWindowUrl).not.be.undefined;
+	});
+
+	it('The runtime version should be set', async () => {
+		const fin = await OpenFinProxy.fin();
+		const version = await fin.System.getVersion();
+		expect(version).to.equal('26.102.71.7');
```
Comment (johnman) [inline how-to/automation-testing/use-launch-external-process-js/tests/index.spec.js:44]: Maybe we should pull the current version from the manifest so it isn't something we have to maintain?

File: how-to/automation-testing/use-window-options-js/automation-hooks.js [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+	globalVars.startTime = Date.now();
+};
+
+exports.teardown = async (globalVars) => {
+	console.log(`Global Teardown`);
+	console.log(`Time taken`, Date.now() - globalVars.startTime);
+};
+
+exports.beforeEach = async (globalVars, currentTestName) => {
+	// console.log("Before", currentTestName);
```
Comment (johnman) [inline how-to/automation-testing/use-window-options-js/automation-hooks.js:13]: remove commented out code or uncomment?

File: how-to/automation-testing/use-window-options-js/tests/index.spec.js [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+
+	it('The url should be set', async () => {
+		providerWindowUrl = await WebDriver.getUrl();
+		expect(providerWindowUrl).not.be.undefined;
+	});
+
+	it('The runtime version should be set', async () => {
+		const fin = await OpenFinProxy.fin();
+		const version = await fin.System.getVersion();
+		expect(version).to.equal('26.102.70.16');
```
Comment (johnman) [inline how-to/automation-testing/use-window-options-js/tests/index.spec.js:44]: should we read the version from the manifest?

**Review by johnman — COMMENTED** [review]
A few comments in there. Generally we try to use Typescript as the default. Was JS used to make it easier for the person reading it (as it is a small test file)?

The examples use xpath and paths. We should note in the readme that this is a basic example and link to other approaches such as the page model for them to consider: https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/

File: how-to/automation-testing/create-window-js/automation-hooks.js (hunk 2 above)
Comment (ameetj85) [reply how-to/automation-testing/create-window-js/automation-hooks.js:1]: Strange but I dont see that in .gitignore. The only entries that I have are: 
chromedriver/
reports/
screenshots/

File: .gitignore (hunk 1 above)
Comment (ameetj85) [reply .gitignore:18]: Strange but I dont see that in .gitignore. The only entries that I have are:
chromedriver/
reports/
screenshots/

File: how-to/automation-testing/create-window-js/tests/index.spec.js [hunk 24]
```diff
@@ -1,20 +1,9 @@
...
-/* eslint-disable no-multiple-empty-lines */
-/* eslint-disable linebreak-style */
-/* eslint-disable no-unused-vars */
-/* eslint-disable linebreak-style */
-const {
-	MouseButton,
-	OpenFinHome,
-	OpenFinNotifications,
-	OpenFinProxy,
+/* eslint linebreak-style: ["error", "windows"] */
```
Comment (obany) [inline how-to/automation-testing/create-window-js/tests/index.spec.js:1]: There is a setting for git and vscode to make sure this works without needing the eslint rule

Comment (adam-saland) [reply how-to/automation-testing/create-window-js/tests/index.spec.js:1]: @obany / @johnman should we add to validate or is only relevant to test framework syntax validation?

Comment (obany) [reply how-to/automation-testing/create-window-js/tests/index.spec.js:1]: So this is just a setting needed for git commits https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings
It just keeps all the code consistent with line endings, which is why the eslint rule is enabled, if you need to disable the rule means your git config is incorrect

File: .gitignore (hunk 1 above)
Comment (obany) [reply .gitignore:18]: It the .gitignore in the root of the repo

Comment (ameetj85) [reply .gitignore:18]: Done

File: how-to/automation-testing/create-window-js/package.json [hunk 25]
```diff
@@ -4,7 +4,9 @@
 	"description": "OpenFin Container Starter Automation Tests",
 	"scripts": {
 		"build": "echo Nothing to build",
-		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json",
+		"setup": "cd ../../../ && npm install && cd how-to/use-a-manifest/create-a-single-page-platform && npm run build",
```
Comment (obany) [inline how-to/automation-testing/create-window-js/package.json:8]: The setup command has the wrong path

File: how-to/automation-testing/.vscode/settings.json [hunk 26]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (obany) [inline how-to/automation-testing/.vscode/settings.json:1]: This file should not be necessary, if you really want these options, then you can add them to your global vscode config

File: how-to/automation-testing/use-launch-external-process-js/package.json [hunk 27]
```diff
@@ -4,7 +4,9 @@
 	"description": "OpenFin Container Starter Automation Tests",
 	"scripts": {
 		"build": "echo Nothing to build",
-		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json",
+		"setup": "cd ../../../ && npm install && cd how-to/use-a-manifest/create-a-single-page-platform && npm run build",
```
Comment (obany) [inline how-to/automation-testing/use-launch-external-process-js/package.json:8]: The setup command has the wrong path

File: how-to/automation-testing/use-window-options-js/package.json [hunk 28]
```diff
@@ -4,7 +4,9 @@
 	"description": "OpenFin Container Starter Automation Tests",
 	"scripts": {
 		"build": "echo Nothing to build",
-		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json"
+		"test-local-mocha": "of-automation http://localhost:5050/manifest.fin.json ./tests/**/*.spec.js --reporter ./mocha.reporter.json",
+		"setup": "cd ../../../ && npm install && cd how-to/use-a-manifest/create-a-single-page-platform && npm run build",
```
Comment (obany) [inline how-to/automation-testing/use-window-options-js/package.json:8]: The setup command has the wrong path

**Review by obany — CHANGES_REQUESTED** [review]

**Review by obany — APPROVED** [review]

**Pull request merged**